zookeeper-server/src/main/java/org/apache/zookeeper/server/quorum/flexible/QuorumVerifier.java

```diff
@@ -39,6 +44,17 @@
     Map<Long, QuorumServer> getVotingMembers();
     Map<Long, QuorumServer> getObservingMembers();
     boolean equals(Object o);
+    /*
+    * Only QuorumOracleMaj will implement these methods. Other class will raise warning if the methods are called and
+    * return false always.
+    * */
+    boolean updateNeedOracle(List<LearnerHandler> forwardingFollowers);
```

Comment:
are you saying that we could add another method called "default()" for example, and make these non-implemented methods call it so that we dont need to copy-paste the dummy implementation?

Like in QuorumMaj.java

boolean updateNeedOracle(List<LearnerHandler> forwardingFollowers) {
      return this.default();
}

Im sorry if I got you wrong.

Reply:
I was referring to the default method feature available since Java 8. See https://www.tutorialspoint.com/java8/java8_default_methods.htm. Basically we just need to move the dummy implementations of the new method into QuorumVerifier, and mark these methods as default.

Reply:
Good point. I will take it. Thank you for bringing it up.